### PR TITLE
[PCM-1387] added screen recording metadata request

### PR DESCRIPTION
**Pull request by jensengar**

### CODE REVIEW

File: src/sessions/screen-recording-session-handler.ts [hunk 1]
```diff
@@ -52,10 +58,31 @@ export default class ScreenRecordingSessionHandler extends BaseSessionHandler {
...
+    await this.updateScreenRecordingMetadatas(session, params.screenRecordingMetadatas);
+
   }
 
   async endSession (conversationId: string, session: IExtendedMediaSession, reason?: Constants.JingleReasonCondition): Promise<void> {
     throw createAndEmitSdkError.call(this.sdk, SdkErrorTypes.not_supported, `sessionType ${this.sessionType} must be ended remotely`, { conversationId });
   }
+
+  updateOutgoingMedia (session: IExtendedMediaSession, options: IUpdateOutgoingMedia): never {
+    this.log('warn', 'Cannot update outgoing media for screen recording sessions', { sessionId: session.id, sessionType: session.sessionType });
```
Comment (djhouseknecht) [inline src/sessions/screen-recording-session-handler.ts:71]: Can we really not update media for SR sessions?

File: src/types/interfaces.ts [hunk 2]
```diff
@@ -809,6 +814,18 @@ export interface VideoMediaSession extends IExtendedMediaSession {
...
+
+export interface ScreenRecordingMetadata {
+  trackId: string;
+  screenId: string;
+  originX: number;
+  originY: number;
+  resolutionX: number;
+  resolutionY: number;
+  primary: boolean;
+  main: boolean;
```
Comment (djhouseknecht) [inline src/types/interfaces.ts:828]: Could we document what these properties are?

**Review by djhouseknecht — CHANGES_REQUESTED** [review]
Few things: 

* We probably want to add the metadatas to the docs. 
* I also noticed that the event name is wrong in the SR docs. It is listed as `sdk.on('sessionStart', ...` but should be `'sessionStarted'`. 
* Will the new required metadata param interfere with the `config.autoAcceptPendingScreenRecordingRequests: true` option? I don't think it should, but I wanted to ask.
* I don't think SR is being used in the SDK yet (hopefully), but would this be considered a breaking change?

**Review by maxwellmooney13 — APPROVED** [review]

File: src/sessions/screen-recording-session-handler.ts (hunk 1 above)
Comment (jensengar) [reply src/sessions/screen-recording-session-handler.ts:71]: well, yes and no. We *could* but there's no precedent for that yet and when we do, we'd need to send new screen metadata.

File: src/types/interfaces.ts (hunk 2 above)
Comment (jensengar) [reply src/types/interfaces.ts:828]: Thanks for reminding me, I need to update some other doc as well.

**Comment by jensengar** [comment]
```
Will the new required metadata param interfere with the config.autoAcceptPendingScreenRecordingRequests: true option? I don't think it should, but I wanted to ask.
```
No it won't have any issue there.

```
I don't think SR is being used in the SDK yet (hopefully), but would this be considered a breaking change?
```
Technically it would probably be considered a breaking change, but I'm going to sneak it through as a patch release since no one is using it.

File: doc/index.md [hunk 3]
```diff
@@ -970,6 +971,28 @@ Params:
...
+      }
+      ```
+      * `trackId: string;` The `MediaStreamTrack.id` associated with this screen.
+      * `screenId: string;` The id associated with the monitor/screen you are recording. This can often be found at `MediaStreamTrack.getSettings().deviceId`.
+      * `originX: number;` The left coordinate for this screen.
+      * `originY: number;` The bottom coordinatefor this screen. *NOTE: Windows and Mac sometimes switch where they reference originY. This property is for playback purposes and a Y coordinate of 0 should always represent the bottom of the screen.
+      * `resolutionX: number` The width of the screen.
+      * `resolutionY: number` The height of the screen.
+      * `primary: boolean` This monitor is the system default/primary monitor where the start bar and/or dock lives.
+      * `main: boolean` This monitor contains the primary application you wish to be recorded.
```
Comment (djhouseknecht) [inline doc/index.md:995]: Did you say you were removing this? Or at least making it non-required?

File: doc/screen-recording.md [hunk 4]
```diff
@@ -74,16 +90,19 @@ which means you will need to combine the streams. Here's an example of that.
 
 ``` ts
 // set up needed events
-sdk.on('sessionStart', async (session) => {
+sdk.on('sessionStarted', async (session) => {
```
Comment (djhouseknecht) [inline doc/screen-recording.md:93]: There are still two instances of `'sessionStart'` above in screen-recording.md. (lines 27 and 58).

File: src/types/interfaces.ts (hunk 2 above)
Comment (djhouseknecht) [reply src/types/interfaces.ts:862]: Same as above.

**Review by djhouseknecht — CHANGES_REQUESTED** [review]

File: doc/index.md [hunk 5]
```diff
@@ -970,6 +971,28 @@ Params:
...
+        resolutionX: number;
+        resolutionY: number;
+        primary: boolean;
+        main: boolean;
+      }
+      ```
+      * `trackId: string;` The `MediaStreamTrack.id` associated with this screen.
+      * `screenId: string;` The id associated with the monitor/screen you are recording. This can often be found at `MediaStreamTrack.getSettings().deviceId`.
+      * `originX: number;` The left coordinate for this screen.
+      * `originY: number;` The bottom coordinatefor this screen. *NOTE: Windows and Mac sometimes switch where they reference originY. This property is for playback purposes and a Y coordinate of 0 should always represent the bottom of the screen.
```
Comment (zservies) [inline doc/index.md:991]: Missing space

**Review by zservies — APPROVED** [review]

File: doc/index.md (hunk 3 above)
Comment (jensengar) [reply doc/index.md]: sorry, I forgot to push my commit.

**Review by djhouseknecht — APPROVED** [review]

**Pull request merged**